tasks/pull-disk-images/pull-disk-images.yaml

```diff
@@ -0,0 +1,88 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: pull-disk-images
+  labels:
+    app.kubernetes.io/version: "0.1.0"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task to pull disk images from quay using oras pull
+  params:
+    - name: snapshotPath
+      type: string
+      description: Path to the JSON string of the Snapshot spec in the data workspace
+    - name: concurrentLimit
+      type: string
+      description: The maximum number of images to be pulled at once
+      default: 3
+  workspaces:
+    - name: data
+      description: |
+        The workspace where the disk images will be written to. This workspace should be large (on the order
+        of 100G).
+  steps:
+    - name: pull-images
+      image:
+        quay.io/konflux-ci/release-service-utils:26f77b0933b19e7318bacf5dee297a6c113bb57e
+      script: |
+        #!/usr/bin/env bash
+        set -eux
+
+        SNAPSHOT_FILE="$(workspaces.data.path)/$(params.snapshotPath)"
+        if [ ! -f "${SNAPSHOT_FILE}" ] ; then
+            echo "No valid snapshot file was provided."
+            exit 1
+        fi
+
+        # Declare a location in the same subdirectory as the pipeline CRs to save the images to
+        DISK_IMAGE_DIR="$(dirname ${SNAPSHOT_FILE})"
+
+        process_component() { # Expected argument is [component json]
+            COMPONENT=$1
+            PULLSPEC=$(jq -r '.repository' <<< "${COMPONENT}")
+            DESTINATION="${DISK_IMAGE_DIR}/$(jq -r '.staged.destination' <<< "${COMPONENT}")/FILES"
+            mkdir -p "${DESTINATION}"
+            DOWNLOAD_DIR=$(mktemp -d)
+            cd $DOWNLOAD_DIR
+            # oras has very limited support for selecting the right auth entry,
+            # so create a custom auth file with just one entry
+            AUTH_FILE=$(mktemp)
+            select-oci-auth "${PULLSPEC}" > "$AUTH_FILE"
+            oras pull --registry-config "$AUTH_FILE" "$PULLSPEC"
+            NUM_MAPPED_FILES=$(jq '.staged.files | length' <<< ${COMPONENT})
+            for ((i = 0; i < $NUM_MAPPED_FILES; i++)) ; do
+                FILE=$(jq -c --arg i "$i" '.staged.files[$i|tonumber]' <<< $COMPONENT)
+                SRC=$(jq -r '.src' <<< $FILE)
+                DST=$(jq -r '.dst' <<< $FILE) # TODO - make apply-mapping expand these
+                gzip -d "${SRC}.gz" || echo "didn't find mapped src file TODO"
```

Comment:
@ralphbean do we want to fail the task if a file is present in the component[*].staged.files but it isn't found when we run `oras pull`?

Reply:
Going to proceed with not failing for now